x-pack/platform/plugins/shared/stack_connectors/server/routes/get_webhook_secret_headers_key.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+import type {
+  IRouter,
+  RequestHandlerContext,
+  KibanaRequest,
+  IKibanaResponse,
+  KibanaResponseFactory,
+} from '@kbn/core/server';
+import type { StartServicesAccessor } from '@kbn/core/server';
+import { ACTION_SAVED_OBJECT_TYPE } from '@kbn/actions-plugin/server';
+
+import type { PluginStartContract as ActionsPluginStartContract } from '@kbn/actions-plugin/server';
+
+import type { EncryptedSavedObjectsPluginStart } from '@kbn/encrypted-saved-objects-plugin/server';
+import { INTERNAL_BASE_STACK_CONNECTORS_API_PATH } from '../../common';
+
+export interface ConnectorsPluginsStart {
+  encryptedSavedObjects: EncryptedSavedObjectsPluginStart;
+  actions: ActionsPluginStartContract;
+}
+
+export interface ConnectorAttributes {
+  secrets?: {
+    secretHeaders?: Array<{ key: string; value: string; type: string }>;
+  };
+  actionTypeId: string;
+}
+
+export const getWebhookSecretHeadersKeyRoute = (
+  router: IRouter,
+  getStartServices: StartServicesAccessor<ConnectorsPluginsStart, unknown>
+) => {
+  router.get(
+    {
+      path: `${INTERNAL_BASE_STACK_CONNECTORS_API_PATH}/{id}/secret_headers`,
+      security: {
+        authz: {
+          enabled: false,
+          reason:
+            'This route is opted out from authorization as returning headers does not require any.',
```

Comment:
nit, but this is misleading.

There is a call to `getActionsClientWithRequest` and `actionsClient.get` does `ensureAuthorized`.

Reply:
+1 let's update the description to capture this.